### PR TITLE
Fix S3 Endpoint or Region configuration.

### PR DESCRIPTION
**Pull request by Antiarchitect**

### CODE REVIEW

**Comment by schabdo** [comment]
Cool, thanks a lot for handing in a fix! It looks like you used the wrong email which has no ECA attached

**Comment by schabdo** [comment]
@hawkbit-bot verify please

**Comment by Antiarchitect** [comment]
Now it looks much better. If we have the endpoint we don't need region at all.

**Comment by Antiarchitect** [comment]
Good news, everyone!
Just tested it with S3 and MinIO - everything works just fine. Updated README with MinIO section. Think you can merge.

File: hawkbit-extension-artifact-repository-s3/src/main/java/org/eclipse/hawkbit/artifact/repository/S3RepositoryAutoConfiguration.java [hunk 1]
```diff
@@ -83,11 +83,10 @@ public ClientConfiguration awsClientConfiguration() {
     public AmazonS3 amazonS3() {
         final AmazonS3ClientBuilder s3ClientBuilder = AmazonS3ClientBuilder.standard()
                 .withCredentials(awsCredentialsProvider()).withClientConfiguration(awsClientConfiguration());
-        if (!StringUtils.isEmpty(region)) {
+        if (!StringUtils.isEmpty(endpoint)) {
+            s3ClientBuilder.withEndpointConfiguration(new EndpointConfiguration(endpoint, ""));
```
Comment (schabdo) [inline hawkbit-extension-artifact-repository-s3/src/main/java/org/eclipse/hawkbit/artifact/repository/S3RepositoryAutoConfiguration.java]: Shouldn't you set region as well if it is provided via property `aws.region`?

**Comment by schabdo** [comment]
> Good news, everyone! 
> Just tested it with S3 and MinIO - everything works just fine. Updated README with MinIO section

Indeed! Thanks a lot for providing a small "how to setup" for MinIO right away

File: hawkbit-extension-artifact-repository-s3/src/main/java/org/eclipse/hawkbit/artifact/repository/S3RepositoryAutoConfiguration.java (hunk 1 above)
Comment (Antiarchitect) [reply hawkbit-extension-artifact-repository-s3/src/main/java/org/eclipse/hawkbit/artifact/repository/S3RepositoryAutoConfiguration.java]: It is not necessary as it goes

Comment (schabdo) [reply hawkbit-extension-artifact-repository-s3/src/main/java/org/eclipse/hawkbit/artifact/repository/S3RepositoryAutoConfiguration.java]: Sorry, for my briefness. I was thinking more about other users/programmers who maybe want to use manually set S3 endpoint and use AWS's `SigV4 signing` (which may require the region to be set). What do you think? Will this work for MinIO as well?

```java
if (!StringUtils.isEmpty(endpoint)) {
   final String signingRegion = StringUtils.isEmpty(region) ? "" : region;
   s3ClientBuilder.withEndpointConfiguration(new EndpointConfiguration(endpoint, signingRegion));
} else if (!StringUtils.isEmpty(region)) {
   s3ClientBuilder.withRegion(region);
}
```

Comment (Antiarchitect) [reply hawkbit-extension-artifact-repository-s3/src/main/java/org/eclipse/hawkbit/artifact/repository/S3RepositoryAutoConfiguration.java]: Probably. I will try to build it and test tomorrow.

**Comment by Antiarchitect** [comment]
@schabdo Sorry for the delay! I've just tested your proposed variant with MinIO (aws.region + aws.s3.endpoint) and true AWS S3 (aws.region only) and all works fine!

**Comment by schabdo** [comment]
No worries, as you see I need sometimes a bit more time to answer, too 😄 
Thanks a lot for testing! I think we're good to go ...

**Pull request merged**